test/tests/providers/capi/capa/platform/atlas/alerting-rules/grafana.rules.test.yml

```
@@ -27,3 +27,68 @@ tests:
               runbook_url: https://intranet.giantswarm.io/docs/support-and-ops/ops-recipes/grafana-down/
               __dashboardUid__: qRQXmRnik
               dashboardQueryParams: "orgId=2"
+  - interval: 1m
+    input_series:
+      # No lag for the first 20 minutes (lag = 0).
+      # Then replication stops working for 50 minutes (lag increases 60 seconds per minute).
+      # Then no lag again (lag = 0).
+      - series: 'cnpg_pg_replication_lag{namespace="monitoring", pod="grafana-postgresql-1", cluster_id="golem", cluster_type="management_cluster", installation="golem"}'
+        values: "0+0x20 0+60x50 0+0x20"
+    alert_rule_test:
+      - alertname: GrafanaPostgresqlReplicationFailure
+        eval_time: 15m
+        exp_alerts: []
+      - alertname: GrafanaPostgresqlReplicationFailure
+        eval_time: 30m
+        exp_alerts: []
+      - alertname: GrafanaPostgresqlReplicationFailure
+        eval_time: 65m
+        exp_alerts:
+          - exp_labels:
+              area: platform
+              cancel_if_outside_working_hours: "true"
+              installation: "golem"
+              pod: grafana-postgresql-1
+              severity: page
+              team: atlas
+              topic: observability
+              cluster_id: "golem"
+            exp_annotations:
+              description: 'grafana-postgresql replication is lagging for pod grafana-postgresql-1.'
+              runbook_url: https://intranet.giantswarm.io/docs/support-and-ops/ops-recipes/grafana-down/
+      - alertname: GrafanaPostgresqlReplicationFailure
+        eval_time: 85m
+        exp_alerts: []
+  - interval: 1m
+    input_series:
+      # For the first 20 minutes there's no archiving failure.
+      # Then, for the next 10 minutes, archiving fails so the last archived time stays the same.
+      # Then archiving works again for 20 minutes so the last failed archived time is updated.
+      - series: 'cnpg_pg_stat_archiver_last_failed_time{namespace="monitoring", pod="grafana-postgresql-1", cluster_id="golem", cluster_type="management_cluster", installation="golem"}'
+        values: "-1+0x20 1200+60x10 -1+0x20"
```

I think the last failed time should remain still, and not get reset?
```suggestion
        values: "-1+0x20 1200+60x10 1800+0x20"
```